packages/components/src/internal/components/editable/EditableGridPanel.tsx

```diff
@@ -143,6 +153,13 @@ export const EditableGridPanel: FC<Props> = memo(props => {
                 extraExportColumns,
                 exportColFilter
             );
+            // timeout needed here to wait for the download to begin. Value of 1000 not chosen scientifically.
+            window.setTimeout(() => {
```

Comment:
Can this be passed in as a callback for exportHandler? Also, the if(setIsDirty && wasDirty) check could be moved up so setTimeout is not called when not dirty

Reply:
> using 1 second sleep is not a robust implementation. In test I've noticed if I navigate away very soon after export, the dirty alert does show but clicking cancel fails to revert the url.

I don't like the one second sleep either, but couldn't find a different way to know it was safe to switch back, so am happy to hear other suggestions.

The clicking of cancel not reverting the URL is a longer-standing issue that I won't address with this PR.